core/signeddata.go

```diff
@@ -181,11 +178,57 @@ func NewPartialVersionedSignedProposal(proposal *eth2api.VersionedSignedProposal
 	}, nil
 }
 
+func NewVersionedSignedBlindedProposal(bp *eth2api.VersionedSignedBlindedProposal) (VersionedSignedProposal, error) {
```

Comment:
IMO these function names do not follow go idioms as they don't return *BlindedProposal and therefore may be confusing in future.
I would rename these to something like `NewVersionedSignedProposalFromBlindedProposal`...

Reply:
> `NewVersionedSignedProposalFromBlindedProposal`

As painful as it looks, you're right :D Will rename.